contrib/hadoop/README.md

```diff
@@ -0,0 +1,21 @@
+Hadoop
+======
+
+This library provides Dataflow sources and sinks to make it possible to read and write
+Apache Hadoop file formats from Dataflow programs.
+
+Currently, only the read path is implemented. A `HadoopFileSource` allows any Hadoop
+`FileInputFormat` to be read as a `PCollection`.
+
+A `HadoopFileSource` can be read from using the `com.google.cloud.dataflow.sdk.io.Read`
+transform. For example:
+
+    HadoopFileSource<K, V> source = HadoopFileSource.from(path, MyInputFormat.class,
+      MyKey.class, MyValue.class);
+    PCollection<KV<MyKey, MyValue>> records = Read.from(mySource);
+
+Alternatively, the `readFrom` method is a convenience method that returns a read
+transform. For example:
+
+    PCollection<KV<MyKey, MyValue>> records = HadoopFileSource.readFrom(path,
+      MyInputFormat.class, MyKey.class, MyValue.class);
```

Comment:
If it is true that this reader is mainly for Hadoop SequenceFiles, it might be nice to add a bit more to the documentation or make the documentation more precise.

I thought based on the name that this Source might read arbitrary record types from HDFS, etc. (But of course you're the Hadoop expert, so you would know better whether this is clear enough.)